app/main.py

```diff
@@ -1,6 +1,26 @@
 from typing import Callable
+from functools import wraps
 
+def cache(func: Callable):
+    """Decorator to cache results of function calls with immutable arguments."""
+    cache_storage = {}
 
-def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    @wraps(func)
+    def wrapper(*args):
+        if args in cache_storage:
```

Comment:
Using `args` as a key in `cache_storage` assumes that all arguments are immutable. While this is true for integers and tuples, be cautious if you plan to extend this to other types that might be mutable, as it could lead to unexpected behavior.